library/network/src/modules/SuSEFirewall.rb

```diff
@@ -83,8 +100,54 @@ def self.get_running_backends
       running_backends
     end
 
+    # Create appropriate firewall instance based on factors such as which backends
+    # are available and/or running/selected.
+    # @return SuSEFirewall2 or SuSEFirewalld instance.
     def self.create
-      SuSEFirewall2.new
+      Yast.import "Mode"
+
+      # Old testsuite
+      if Mode.testsuite
+        # For the old testsuite, always generate SF2 instance. FirewallD tests
+        # will be committed later on but they will only affect the new
+        # testsuite
+        SuSEFirewall2.new
+      else
+        selected_backend = nil
+        enabled_backends = self.get_enabled_backends
+        running_backends = self.get_running_backends
+
+        begin
+          # Only one running backend is permitted.
+          raise ArgumentError if running_backends.size > 1
```

Comment:
It's OK to give up on 2 backends at once, but it is not an ArgumentError ("you gave a method an argument that is not allowed"), is it? I would expect a (subclass of) StandardError.